src/Symfony/Component/Serializer/Tests/Encoder/XmlEncoderTest.php

```diff
@@ -465,6 +473,23 @@ public function testDecodeEmptyXml()
         $this->encoder->decode(' ', 'xml');
     }
 
+    protected function getXmlPISource()
+    {
+        return '<?xml version="1.0"?>'."\n".
+            '<?xml-stylesheet type="text/xsl" href="/xsl/xmlverbatimwrapper.xsl"?>'.
+            '<?display table-view?>'.
+            '<?sort alpha-ascending?>'.
+            '<response>'.
+            '<foo>foo</foo>'.
+            '<?textinfo whitespace is allowed ?>'.
+            '<bar>a</bar><bar>b</bar>'.
+            '<baz><key>val</key><key2>val</key2><item key="A B">bar</item>'.
+            '<item><title>title1</title></item><?item ignore-title ?><item><title>title2</title></item>'.
+            '<Barry><FooBar id="1"><Baz>Ed</Baz></FooBar></Barry></baz>'.
+            '<qux>1</qux>'.
+            '</response><?instruction <value> ?>'."\n";
```

Comment:
I also prefer the nowdoc syntax as suggested by @greg0ire. As an interesting side effect, it allows PHPStorm to highlight embedded document.